### PR TITLE
Seed Contract types, Departments & Job titles

### PR DESCRIPTION
**Pull request by jackcarlisle**
# ready for review

- [x] seeds data for Contract types, Departments & Job titles

#31

### CODE REVIEW

File: lib/ev2/production/department/department_api.ex [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+          iex> get_departments_with_jobs
+          %{
+            Accounts: %{
+              "Accountant",
+              "Accounts Manager"
+            }
+          }
+
+      """
+      def get_departments_with_jobs do
```
Comment (JMurphyWeb) [inline lib/ev2/production/department/department_api.ex]: @jackcarlisle doesn't this function do what we wanted for the JSON more or less?

What is the inner map? I don't understand how there are no key-values?

```
 %{
 +              "Accountant",
 +              "Accounts Manager"
 +            }
```

File: priv/repo/migrations/20171116144616_create_contract_types.exs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+defmodule Ev2.Repo.Migrations.CreateContractTypes do
+  use Ecto.Migration
+
+  def change do
+    create table(:contract_types) do
+      add :name, :string
+
+      timestamps()
```
Comment (JMurphyWeb) [inline priv/repo/migrations/20171116144616_create_contract_types.exs]: Can we remove timestamps ?

File: priv/repo/migrations/20171116145029_create_job_titles.exs [hunk 3]
```diff
@@ -6,10 +6,8 @@ defmodule Ev2.Repo.Migrations.CreateJobTitles do
       add :name, :string
       add :department_id, references(:departments, on_delete: :nothing)
       add :contract_type_id, references(:contract_types, on_delete: :nothing)
-
-      timestamps()
     end
-
+    create unique_index(:job_titles, [:name, :department_id])
```
Comment (JMurphyWeb) [inline priv/repo/migrations/20171116145029_create_job_titles.exs]: Can we add a unique index in the department table for department name?

```
    create unique_index(:departments, [:name], name: "unique_department")
```

Also adding a 'name' to the unique index above will make it easier/clearer to reference if we want to in the future, would you mind adding one? Something like: `name: "unique_job_in_department"`

File: lib/ev2_web/controllers/dashboard_controller.ex [hunk 4]
```diff
@@ -25,7 +25,10 @@ defmodule Ev2Web.DashboardController do
         vehicle_details_complete? =
           case company_user? do
             false -> false
-            true -> Production.get_vehicle_by_user!(user).completed?
+            true ->
+              user
+              |> Production.get_vehicle_by_user!()
+              |> Production.vehicle_details_complete?()
```
Comment (JMurphyWeb) [inline lib/ev2_web/controllers/dashboard_controller.ex]: this gets refactored out, so maybe worth merging that pr first?

File: lib/ev2/production/vehicle/vehicle_api.ex [hunk 5]
```diff
@@ -160,6 +160,23 @@ defmodule Ev2.Production.VehicleAPI do
...
+
+      """
+      def vehicle_details_complete?(vehicle) do
+        vehicle.name != ""
+        && vehicle.address_1 != ""
+        && vehicle.address_city != ""
+        && vehicle.address_postcode != ""
+        && vehicle.address_country != ""
+      end
+
```
Comment (JMurphyWeb) [inline lib/ev2/production/vehicle/vehicle_api.ex]: I moved this function into view helpers in refactor dashboard. Maybe worth getting that PR in first?

File: lib/ev2/production/department/department_api.ex (hunk 1 above)
Comment (jackcarlisle) [reply lib/ev2/production/department/department_api.ex]: So that was a typo, should be a list

**Pull request merged**